TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementCheckers.cs

```diff
@@ -16,10 +18,120 @@ public static class SavepointStatementChecker
+                if (tokenValue.Type != LiteralTokenValueType.Decimal) return;
+                var decimalLiteral = (DecimalLiteralTokenValue) tokenValue;
+                bool isInteger = Math.Abs(decimalLiteral.Number % 1) < double.Epsilon;
+                if (!isInteger)
+                {
+                    DiagnosticUtils.AddError(alterSequenceStatement,
+                        "The maxValue decimal should be without non-zero digits in the right of the decimal point",
+                        context);
+                }
+
+                if (alterSequenceStatement.MinValue != null)
+                {
+                    var minTokenValue = alterSequenceStatement.MinValue.Literal.LiteralValue;
+                    if (minTokenValue.Type != LiteralTokenValueType.Decimal) return;
+                    var minDecimalLiteral = (DecimalLiteralTokenValue) minTokenValue;
+                    if (minDecimalLiteral.Number > decimalLiteral.Number)
+                    {
+                        DiagnosticUtils.AddError(alterSequenceStatement,
+                            "The maxValue must be greater than or equal to the minimum value.",
+                            context);
+                    }
+                }
+            }
+
+            if (alterSequenceStatement.RestartValue != null)
+            {
+                var tokenValue = alterSequenceStatement.RestartValue.Literal.LiteralValue;
+                if (tokenValue.Type != LiteralTokenValueType.Decimal) return;
+                var decimalLiteral = (DecimalLiteralTokenValue) tokenValue;
+                bool isInteger = Math.Abs(decimalLiteral.Number % 1) < double.Epsilon;
+                if (!isInteger)
+                {
+                    DiagnosticUtils.AddError(alterSequenceStatement,
+                        "The restartValue decimal should be without non-zero digits in the right of the decimal point",
+                        context);
+                }
+            }
+
+            if (alterSequenceStatement.IncrementValue != null)
+            {
+                var tokenValue = alterSequenceStatement.IncrementValue.Literal.LiteralValue;
+                if (tokenValue.Type != LiteralTokenValueType.Decimal) return;
+                var decimalLiteral = (DecimalLiteralTokenValue) tokenValue;
+                bool isInteger = Math.Abs(decimalLiteral.Number % 1) < double.Epsilon;
+                if (!isInteger)
+                {
+                    DiagnosticUtils.AddError(alterSequenceStatement,
+                        "The IncrementValue decimal should be without non-zero digits in the right of the decimal point",
+                        context);
+                }
+            }
+
+            if (alterSequenceStatement.CacheSize != null)
+            {
+                var tokenValue = alterSequenceStatement.CacheSize.Literal.LiteralValue;
+                var integerLiteral = (IntegerLiteralTokenValue) tokenValue;
+                if (integerLiteral.HasSign)
```

Comment:
No. `HasSign` property indicates whether the sign has been explicitly written, to check whether the number is negative or not, you have to test the value. That's why it will become redundant with the check below.